CHANGELOG.md

```diff
@@ -7,10 +7,18 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 
 ### Added
+- **CUMULUS-851** Added workflow lambda versioning feature to allow in-flight workflows to use lambda versions that were in place when a workflow was initiated
+    - Updated Kes custom code to remove logic that used the CMA file key to determine template compilation logic.  Instead, utilize a `customCompilation` template configuration flag to indicate a template should use Cumulus's kes customized methods instead of 'core'.
+	- Added `useWorkflowLambdaVersions` configuration option to enable the lambdaVersioning feature set.   **This option is set to true by default** and should be set to false to disable the feature.
+	- Added uniqueIdentifier configuration key to S3 sourced lambdas to optionally support S3 lambda resource versioning within this scheme. This key must be unique for each modified version of the lambda package and must be updated in configuration each time the source changes.
+    - Added a new nested stack template that will create a `LambdaVersions` stack that will take lambda parameters from the base template, generate lambda versions/aliases and return outputs with references to the most 'current' lambda alias reference, and updated 'core' template to utilize these outputs (if `useWorkflowLambdaVersions` is enabled).
+      - Updated Kes custom code to add `oldLambdaInjection` template configuration key to trigger addition of oldLambdas key to template config object.  This option requres `useWorkflowLambdaVersions` to be enabled, a `workflowLambdas` key to be configured in the template configuration, and is not intended to be user-facing.
+      - Updated Kes custom code to add `injectWorkflowLambdaAliases` template configuration key to trigger substitiution of expected 'outputs' from the LambdaVersions nested stack. This option requires `useWorkflowLambdaVersions` to be enabled, a `workflowLambdas` key to be configured in the template configuration, and is not intended to be user-facing.
```

Comment:
Do we really need this conditional at this point if it is not intended to be user-facing?

Reply:
I think so, yes.    We talked about this a bit, but for posterity, the template config options are being used to identify which logic branch a template should take, and simplification of this likely requires a refactor/feature request or two on kes core, or hard-coded logic that maps to template names.      There is a follow on refactor ticket that should address this.